dev/devicelab/bin/tasks/ios_app_with_watch_companion.dart

```diff
@@ -0,0 +1,219 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:flutter_devicelab/framework/framework.dart';
+import 'package:flutter_devicelab/framework/utils.dart';
+import 'package:path/path.dart' as path;
+
+Future<void> main() async {
+  await task(() async {
+    section('Copy test Flutter App with WatchOS Companion');
+
+    String watchDeviceID;
+    String phoneDeviceID;
+    final Directory tempDir = Directory.systemTemp
+        .createTempSync('ios_app_with_watch_companion_test');
+    final Directory projectDir =
+        Directory(path.join(tempDir.path, 'app_with_companion'));
+    try {
+      mkdir(projectDir);
+      recursiveCopy(
+        Directory(path.join(flutterDirectory.path, 'dev', 'integration_tests',
+            'ios_app_with_watch_companion')),
+        projectDir,
+      );
+
+      section('Create release build');
+
+      await inDirectory(projectDir, () async {
+        await flutter(
+          'build',
+          options: <String>['ios', '--no-codesign'],
+        );
+      });
+
+      final bool appReleaseBuilt = exists(Directory(path.join(
+        projectDir.path,
+        'build',
+        'ios',
+        'iphoneos',
+        'Runner.app',
+      )));
+
+      if (!appReleaseBuilt) {
+        return TaskResult.failure(
+            'Failed to build flutter iOS app with WatchOS companion in release mode.');
+      }
+
+      section('Create debug build');
+
+      await inDirectory(projectDir, () async {
+        await flutter(
+          'build',
+          options: <String>['ios', '--debug', '--no-codesign'],
+        );
+      });
+
+      final bool appDebugBuilt = exists(Directory(path.join(
+        projectDir.path,
+        'build',
+        'ios',
+        'iphoneos',
+        'Runner.app',
+      )));
+
+      if (!appDebugBuilt) {
+        return TaskResult.failure(
+            'Failed to build flutter iOS app with WatchOS companion in debug mode.');
+      }
+
+      section('Create build for a simulator device');
+
+      // Create iOS simulator devices.
+      phoneDeviceID = await eval(
+        'xcrun',
+        <String>[
+          'simctl',
+          'create',
+          'TestFlutteriPhoneWithWatch',
+          'com.apple.CoreSimulator.SimDeviceType.iPhone-11',
+          'com.apple.CoreSimulator.SimRuntime.iOS-13-2'
```

Comment:
Leave off the `com.apple.CoreSimulator.SimRuntime.iOS-13-2` and  `com.apple.CoreSimulator.SimRuntime.watchOS-6-1` runtime IDs on this and the watchOS `simctl create`.  It fails on my machine with iOS-13-4:
```
[ios_app_with_watch_companion] [STDOUT] Executing: xcrun simctl create TestFlutteriPhoneWithWatch com.apple.CoreSimulator.SimDeviceType.iPhone-11 com.apple.CoreSimulator.SimRuntime.iOS-13-2
...
[ios_app_with_watch_companion] [STDOUT] stderr: Invalid runtime: com.apple.CoreSimulator.SimRuntime.iOS-13-2
[ios_app_with_watch_companion] [STDOUT] "xcrun" exit code: 161
```
However it succeeded when I left removed those arguments:
```
Task result:
{
  "success": true,
  "data": null,
  "benchmarkScoreKeys": []
}
```
```
$ xcrun simctl create -h   
Usage: simctl create <name> <device type id> [<runtime id>]
...
 If no runtime is specified the newest runtime compatible with the device type is chosen.
```

Reply:
Thanks for pointing this out, this is a much better solution and I never liked hardcoding the runtime version anyway. Apparently my Xcode version was too old and required the runtime parameter.

Reply:
The hardcoded runtime parameter has been removed. :-)